matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/store/db/RealmCryptoStore.kt

```diff
@@ -389,6 +390,29 @@ internal class RealmCryptoStore @Inject constructor(
         }
     }
 
+    override fun saveBackupRecoveryKey(recoveryKey: String?, version: String?) {
+        doRealmTransaction(realmConfiguration) { realm ->
+            realm.where<CryptoMetadataEntity>().findFirst()?.apply {
+                keyBackupRecoveryKey = recoveryKey
+                keyBackupRecoveryKeyVersion = version
+            }
+        }
+    }
+
+    override fun getKeyBackupRecoveryKeyInfo(): SavedKeyBackupKeyInfo? {
+        return doRealmQueryAndCopy(realmConfiguration) { realm ->
+            realm.where<CryptoMetadataEntity>().findFirst()
+        }?.let {
+            val key = it.keyBackupRecoveryKey
+            val version = it.keyBackupRecoveryKeyVersion
+            if (!key.isNullOrBlank() && !version.isNullOrBlank()) {
+                SavedKeyBackupKeyInfo(recoveryKey = key, version = version)
+            } else {
+                null
```

Comment:
We can take advantage of takeIf() to remove ugly "null".

Reply:
takeif is execute after, i'll have to default (?: "")